clippy_lints/src/bool_assert_comparison.rs

```diff
@@ -90,13 +103,28 @@ impl<'tcx> LateLintPass<'tcx> for BoolAssertComparison {
                         }
 
                         let non_eq_mac = &mac[..mac.len() - 3];
+                        let expr_string = if lit_value ^ is_eq {
+                            format!("!({})", source::snippet(cx, other_expr.span, ""))
```

Comment:
`snippet` should be directly imported. Use `Sugg::hir_with_applicability` and `Sugg::maybe_par`. The default value should be `".."` instead of `""`.